### DISCUSSION TITLE
espeak-ng and espeak_wrapper on windows, omitted letters

### DISCUSSION THREAD

**Discussion by houkman** [coqui-ai/TTS#3761] [General]
I noticed that [espeak-ng](https://github.com/espeak-ng/espeak-ng) (1.52-dev) launched from the console produces wrong output (phonemization and sound). Some letters are completely omitted. The problem is already known: [https://sourceforge.net/p/espeak/bugs/125/](https://sourceforge.net/p/espeak/bugs/125/). Changing the character encoding in the console to UTF-8 (`chcp 65001`) does not help. The same thing happens with [espeak_wrapper.py](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/utils/text/phonemizers/espeak_wrapper.py). For example: for polish word - "źrebię" ("foal" - in english), after running in python:

```
>>> exec(open("espeak_wrapper.py", encoding="utf-8").read(), globals())
>>> from TTS.tts.utils.text.phonemizers import ESpeak
>>> phonemizer = ESpeak("pl")
>>> phonemizer.phonemize("źrebię", separator="")
```

result is: `"rˈɛbʲi"` but it should be: `"ʑrˈɛbjɛ"`
(referring to [https://odo.lv/Espeak?eslang=-vpl&speed=160&output=-w&debug=-x&text=źrebię&formmethod=get](https://odo.lv/Espeak?eslang=-vpl&speed=160&output=-w&debug=-x&text=źrebię&formmethod=get))

The letters "ź" and "ę" are completely omitted. I think this also applies to languages ​​other than polish. I'm not a python programmer, but checking [def _espeak_exe](https://github.com/coqui-ai/TTS/blob/dbf1a08a0d4e47fdad6172e433eeb34bc6b13b4e/TTS/tts/utils/text/phonemizers/espeak_wrapper.py#L47) in [espeak_wrapper.py](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/utils/text/phonemizers/espeak_wrapper.py) it looks like the text that is phonemized is readed from the console. I think this is the wrong approach and on windows it should be read from a UTF-8 encoded file, example:

`espeak-ng -vpl --ipa -f utf8_text.txt`

I modified espeak_wrapper.py ([def phonemize_espeak)](https://github.com/coqui-ai/TTS/blob/dbf1a08a0d4e47fdad6172e433eeb34bc6b13b4e/TTS/tts/utils/text/phonemizers/espeak_wrapper.py#L156) for my needs from:
```
if tie:
	args.append("--tie=%s" % tie)

args.append(text)
# compute phonemes
```
to:
```
if tie:
	args.append("--tie=%s" % tie)

UTF8filepath = "p:\\temp\\phonemes.txt"
UTF8file = open(UTF8filepath, "w", encoding="utf-8")
UTF8file.write(text)
UTF8file.close()

args.append("-f")
args.append(UTF8filepath)
# compute phonemes
```
and now phonemize works properly. As I said, I'm not a python programmer, everything I did is based on google. Someone should take a closer look at this. I'm on windows 10, python v3.11, coqui TTS-0.22.0.

**Comment by eginhard** [comment]
This repo is not maintained anymore, but I just fixed this in [our fork](https://github.com/idiap/coqui-ai-TTS) (available via `pip install coqui-tts`) by having Espeak read the input from a temporary file. Thank you for spotting and reporting it!
